src/wp-includes/post-template.php

```diff
@@ -1992,15 +1992,15 @@ function wp_list_post_revisions( $post = 0, $type = 'all' ) {
 		_deprecated_argument( __FUNCTION__, '3.6.0' );
 	}
 
-	$revisions = wp_get_post_revisions( $post->ID );
+	$revisions = wp_get_post_revisions( $post->ID, array( 'fields' => 'ids' ) );
```

Comment:
Due to the call to `wp_is_post_autosave()` later in this function, limiting the query to IDs only will cause an increase in the number of DB queries. As a POC test:

```php
/**
 * @ticket 61849
 */
public function test_wp_list_post_revisions_db_queries() {
	$post    = get_default_post_to_edit( 'post', true );
	$post_id = $post->ID;

	// Create a few revisions.
	foreach ( range( 1, 5 ) as $i ) {
		wp_update_post(
			array(
				'post_status'  => 'publish',
				'post_content' => 'Content ' . $i,
				'ID'           => $post_id,
			)
		);
	}

	// Flush the cache.
	wp_cache_flush();

	// Get number of queries before calling wp_list_post_revisions.
	$queries = get_num_queries();
	get_echo( 'wp_list_post_revisions', array( $post_id ) );
	$number_of_queries = get_num_queries() - $queries;

	// Expect four queries for the revisions.
	$this->assertSame( 4, $number_of_queries );
}
```

On trunk the test passes, on this branch it fails:

```
$ ./vendor/bin/phpunit --group 61849
Installing...
Running as single site... To run multisite, use -c tests/phpunit/multisite.xml
Not running ajax tests. To execute these, use --group ajax.
Not running ms-files tests. To execute these, use --group ms-files.
Not running external-http tests. To execute these, use --group external-http.
PHPUnit 9.6.19 by Sebastian Bergmann and contributors.

Warning:       Your XML configuration validates against a deprecated schema.
Suggestion:    Migrate your XML configuration using "--migrate-configuration"!

F                                                                   1 / 1 (100%)

Time: 00:00.160, Memory: 199.00 MB

There was 1 failure:

1) Tests_Post_Revisions::test_wp_list_post_revisions_db_queries
Failed asserting that 8 is identical to 4.

/vagrant/wordpress-develop/tests/phpunit/tests/post/revisions.php:48
```

Reply:
@peterwilsoncc Thank you for the POC test. I run the same test and yes, more queries here. I will look to revert the commit.